GPflowOpt/acquisition/acquisition.py

```diff
@@ -56,10 +59,11 @@ def _optimize_models(self):
         """
         Optimizes the hyperparameters of all models that the acquisition function is based on.
 
-        It is called after initialization and set_data(), and before optimizing the acquisition function itself.
+        It is called automatically after initialization and each time set_data() is called.
+        When using the high-level :class:`..BayesianOptimizer` class calling set_data() is taken care of.
```

Comment:
I hope I got this right. The old explanation did not seem quite right to me. The part about the high-level class could be removed as that will be explained in BayesianOptimizer itself.